Core/src/org/sleuthkit/autopsy/centralrepository/AddEditCentralRepoCommentAction.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.centralrepository;
+
+import java.awt.event.ActionEvent;
+import java.util.logging.Level;
+import javax.swing.AbstractAction;
+import org.openide.util.NbBundle.Messages;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.centralrepository.datamodel.CorrelationAttribute;
+import org.sleuthkit.autopsy.centralrepository.datamodel.EamArtifactUtil;
+import org.sleuthkit.autopsy.centralrepository.datamodel.EamDb;
+import org.sleuthkit.autopsy.centralrepository.datamodel.EamDbException;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * An AbstractAction to manage adding and modifying a Central Repository file
+ * instance comment.
+ */
+public final class AddEditCentralRepoCommentAction extends AbstractAction {
+
+    private static final Logger logger = Logger.getLogger(AddEditCentralRepoCommentAction.class.getName());
+
+    private boolean addToDatabase;
+    private CorrelationAttribute correlationAttribute;
+    String title;
+
+    /**
+     * Private constructor to create an instance given a CorrelationAttribute.
+     *
+     * @param correlationAttribute The correlation attribute to modify.
+     * @param title                The text for the menu item.
+     */
+    private AddEditCentralRepoCommentAction(CorrelationAttribute correlationAttribute, String title) {
+        super(title);
+        this.title = title;
+        this.correlationAttribute = correlationAttribute;
+    }
+
+    /**
+     * Private constructor to create an instance given an AbstractFile.
+     *
+     * @param file  The file from which a correlation attribute to modify is
+     *              derived.
+     * @param title The text for the menu item.
+     */
+    private AddEditCentralRepoCommentAction(AbstractFile file, String title) throws EamDbException, NoCurrentCaseException, TskCoreException {
```

Comment:
Looking at this again, I don't think anything in here actually throws an exception and you can get rid of the entire "throws" part. This should allow you to remove the exception handling from createAddEditCentralRepoCommentAction, and possibly get rid of the new exception class if it isn't used anywhere else.